src/decoder/mod.rs

```diff
@@ -425,9 +422,16 @@ impl<R: Read> Reader<R> {
         len + match extra { 0 => 0, _ => 1 }
     }
 
-    fn allocate_out_buf(&mut self) {
+    fn allocate_out_buf(&mut self) -> Result<(), DecodingError> {
         let width = get_info!(self).width;
-        self.processed = vec![0; self.line_size(width)]
+        let bytes = self.limits.bytes;
+        if bytes < self.line_size(width) {
+            // FIXME: DecodingError::Other is used for backwards compatibility.
```

Comment:
This reasoning doesn't make much sense, as this is a backwards incompatible commit set–a public member of `Limits` was renamed from `pixels` to `bytes`.

Reply:
I simply copied code from the previous limit stuff, but yeah I realize that this doesn't really make sense. I will add a new error variant.